run.c

```diff
@@ -37,37 +37,93 @@ enum ActionType {
     TERMINATE = 5,
     IOCTL = 6,
 };
-static enum ActionType human_interaction();
-static int find_next_sync_point(int ti_nextoutput);
-static void compute_accelerations(void);
+static enum ActionType human_interaction(double lastPMlength);
+static int should_we_timeout(double lastPMlength);
+static void compute_accelerations(int is_PM);
 static void update_IO_params(const char * ioctlfname);
-static void every_timestep_stuff(int NumForces);
-static void write_cpu_log(void);
-
+static void every_timestep_stuff(int NumForces, int NumCurrentTiStep);
+static void write_cpu_log(int NumCurrentTiStep);
 
 void run(void)
 {
     enum ActionType action = NO_ACTION;
+    /*Number of timesteps performed this run*/
+    int NumCurrentTiStep = 0;
+
+    /*To compute the wall time between PM steps*/
+    double lastPM = All.CT.ElapsedTime;
+    double lastPMlength = 0.03*All.TimeLimitCPU;
 
     walltime_measure("/Misc");
 
-    write_cpu_log(); /* produce some CPU usage info */
+    write_cpu_log(NumCurrentTiStep); /* produce some CPU usage info */
+    unsigned int Ti_nextoutput = find_next_outputtime(All.Ti_Current);
 
     do /* main loop */
     {
         /* find next synchronization point and the timebins active during this timestep.
          * If needed, this function will also write an output file
          * at the desired time.
          */
-        All.Ti_Current = find_next_sync_point(All.Ti_nextoutput);
+        All.Ti_Current = find_next_kick(All.Ti_Current);
```

Comment:
The main problem actually is the calls to drift_particle() - because we don't know yet what form the long/short split will take I have been avoiding removing them. I think once we eventually settle on that algorithm we can remove All.Ti_Current. I could also hide it behind a DEBUG flag fairly easily if you wanted that.